src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseGradingService.java

```diff
@@ -374,15 +387,38 @@ private void createFeedbackForNotExecutedTests(Result result, Set<ProgrammingExe
         result.addFeedbacks(feedbacksForNotExecutedTestCases);
     }
 
+    private boolean createFeedbackForDuplicateTests(Result result, ProgrammingExercise programmingExercise) {
+        Set<String> uniqueFeedbackNames = new HashSet<>();
+        // Find duplicate test cases from feedback which is automatic feedback
+        Set<String> duplicateFeedbackNames = result.getFeedbacks().stream()
+                .filter(feedback -> !feedback.isStaticCodeAnalysisFeedback() && FeedbackType.AUTOMATIC.equals(feedback.getType())).map(Feedback::getText)
+                // Set.add() returns false if the element was already in the set, this is how we find all duplicates
+                .filter(fbName -> !uniqueFeedbackNames.add(fbName)).collect(Collectors.toSet());
```

Comment:
fbName --> feedbackName
(please don't use abbreviations)

Reply:
done in 8f8252aea4715d041d5c3aad1c820371083b6655